### ISSUE TITLE
Rule proposal: enforce it() description to start with "should"

### ISSUE THREAD

**Issue by alecxe** [lo1tuma/eslint-plugin-mocha#66]
Would it make sense to enforce having descriptive test names by warning if `it()` description does not start with "should". Example of violation:

```
it("I want to test something", function () {
  // ...
});
```

Valid usage:

```
it("should test something", function () {
  // ...
});
```

Thanks.

**Comment by lo1tuma** [comment]
Thanks for the issue.

I like the idea and I think it would make sense as a stylistic rule. Do you have an idea how a ruleset for meaningful test descriptions could look like?

**Comment by alecxe** [comment]
@lo1tuma I would probably just start with simply checking for the `should` at the beginning of `it()` descriptions. We may think about doing something similar on a `describe` level, but I am not exactly sure what can we enforce there besides checking that there is a non-empty description..

What do you think? Thanks.

By the way, I can work on that and send PR(s).

**Comment by benmosher** [comment]
FWIW, I always try _not_ to use "should". i.e. I try to say

```js
it("passes this test", () => ...)
```

instead of 

```js
it("should pass this test", () => ...)
```

since the `should` is arguably implicit, and thus a waste of 7 chars. 🤓

I probably would not use a rule to enforce test description style, though, so don't take me too seriously. But I can imagine someone else wanting to use a `never` version of this rule.

**Comment by lo1tuma** [comment]
@alecxe Sounds good. Let’s start simple, we can always extend the rule later.

@benmosher Yeah, in some projects I also completely avoid "should". Maybe an option would make sense to forbid the use of "should".

**Comment by alecxe** [comment]
How about this: we implement this rule with a configurable parameter - a regular expression that an `it()` description should match against? By default, we would configure it as `^should`.

We can also do the same for `describe` but have this rule disabled by default.

What do you think? Thanks.

**Comment by lo1tuma** [comment]
Sounds good to me.
